src/channel_state.ts

```diff
@@ -269,6 +284,58 @@ export class ChannelState<
     }
   }
 
+  /**
+   * setPinnedMessages - updates messages in pinned_messages property. See setPinnedMessages for details
+   *
+   * @param {Array<MessageResponse<AttachmentType, ChannelType, CommandType, MessageType, ReactionType, UserType>>} newMessages A list of messages
+   *
+   */
+  setPinnedMessages(
+    newMessages: MessageResponse<
+      AttachmentType,
+      ChannelType,
+      CommandType,
+      MessageType,
+      ReactionType,
+      UserType
+    >[],
+  ) {
+    for (let i = 0; i < newMessages.length; i += 1) {
+      this.setPinnedMessage(newMessages[i]);
+    }
+  }
+
+  /**
+   * setPinnedMessage - update message in pinned_messages property. The messages will be added to pinned_messages if it
+   * is pinned and removed otherwise
+   *
+   * @param {MessageResponse<AttachmentType, ChannelType, CommandType, MessageType, ReactionType, UserType>} newMessage message to update
+   *
+   */
+  setPinnedMessage(
+    newMessage: MessageResponse<
+      AttachmentType,
+      ChannelType,
+      CommandType,
+      MessageType,
+      ReactionType,
+      UserType
+    >,
+  ) {
+    const message = this.messageToImmutable(newMessage);
```

Comment:
```suggestion
    const message = this.messageToImmutable(pinnedMessage);
```
better to move this inside the if case